lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -246,23 +249,39 @@ def get(self, request, *args, **kwargs):
             access_expiration = get_access_expiration_data(request.user, course_overview)
             cert_data = get_cert_data(request.user, course, enrollment.mode) if is_enrolled else None
 
-            # Only show the set course goal message for enrolled, unverified
-            # users in a course that allows for verified statuses.
-            is_already_verified = CourseEnrollment.is_enrolled_as_verified(request.user, course_key)
-            if not is_already_verified and has_course_goal_permission(request, course_key_string,
-                                                                      {'is_enrolled': is_enrolled}):
-                course_goals = {
-                    'goal_options': valid_course_goals_ordered(include_unsure=True),
-                    'selected_goal': None
-                }
-
-                selected_goal = get_course_goal(request.user, course_key)
-                if selected_goal:
-                    course_goals['selected_goal'] = {
-                        'key': selected_goal.goal_key,
-                        'text': get_course_goal_text(selected_goal.goal_key),
+            if COURSE_GOALS_NUMBER_OF_DAYS_GOALS.is_enabled():
+                if (is_enrolled and ENABLE_COURSE_GOALS.is_enabled(course_key)
+                        and settings.FEATURES.get('ENABLE_COURSE_GOALS')):
```

Comment:
Wait what? This is *also* a classic feature flag? That is too many flags. If they both continue to exist, they should be combined into like a `are_course_goals_enabled()` method that returns the combo of the flag and the feature setting. But really...  we should just remove one of them. Presumably the feature setting, since it looks like we wanted to allow course opt-outs, which we can just do with the waffle... 

Looks like the legacy code path used `has_course_goal_permission` for the utility method that checked both toggles. I'd recommend you make a parallel method for the new feature - it could even check all three toggles maybe, if that made some code simpler.

We changed our plans for the goal feature scope to include audit users too? Neat.

Reply:
Ah right, we _only_ showed to audit (not excluded them) and it was because our previous best goal was verifying.